pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -184,22 +184,20 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 		minVersion = &d.Version
 	}
 
-	isServiceReady, err := services.IsServiceReady(d.Client, *internalService)
```

Comment:
Just realized that we should maybe do a similar change in the [autoscaling client](https://github.com/elastic/cloud-on-k8s/blob/634ccd111feddf70f3bcaf21db9ecce990bc9d94/pkg/controller/autoscaling/elasticsearch/driver.go#L101) and remove `services.IsServiceReady`?